distributed/worker.py

```diff
@@ -268,7 +275,8 @@ def heartbeat(self):
                     in_memory=len(self.data),
                     ready=len(self.ready),
                     in_flight=len(self.in_flight_tasks),
-                    **self.monitor.recent())
+                    custom_metrics_names=list(custom_metrics_names_no_overlap),
+                    **metrics)
```

Comment:
I wonder if maybe we should just do `metrics=metrics` here, and avoid the ambiguity of the `**kwargs`.  This would require changing the `**info` in `Scheduler.heartbeat_worker` as well.

Reply:
Thoughts?

Reply:
Maybe this is a good idea ... at the moment it is not that clear (unless you know the signature of course) which are variables of the function and which ones are metrics that end up in `worker_state.info`.

I noticed also that the current code is not robust against using other names like `in_flight`, `ready`, etc ... as names of custom metrics, I need to fix that.

If we go down the route of getting rid of the `**kwargs` we probably want to do the same for `add_worker`:
https://github.com/dask/distributed/blob/a1d4a9d698d760496e06c0790cf4a617065ccbb3/distributed/scheduler.py#L1228-L1230

Reply:
Agreed

Reply:
Is this something that you would be willing to do within this PR?

Reply:
I reckon it is quicker to do this in the same PR.